tests/app/questionnaire/test_questionnaire_store_updater.py

```diff
@@ -1404,10 +1429,180 @@ def test_dependent_sections_added_dependant_block_removed(
+                "section-2": OrderedSet(
+                    ["section-3", "section-4", "section-5", "section-6"]
+                ),
+                "section-3": OrderedSet(["section-4", "section-5", "section-7"]),
+            }
+        )
+    )
+    mock_schema.get_repeating_list_for_section.return_value = False
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "section-1",
+                "block_ids": ["block-1", "block-2"],
+                "status": "IN PROGRESS",
+            },
+            {
+                "section_id": "section-2",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-3",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-4",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-5",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-6",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-7",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+        ],
+    )
+
+    questionnaire_store_updater = get_questionnaire_store_updater(
+        current_location=current_location,
+        progress_store=progress_store,
+        router=mock_router,
+        schema=mock_schema,
+    )
+
+    mocker.patch(
+        "app.questionnaire.questionnaire_store_updater.QuestionnaireStoreUpdater.get_chronological_section_dependents",
+        return_value=[
+            DependentSection(
+                section_id="section-1", list_item_id=None, is_complete=True
+            ),
+            DependentSection(
+                section_id="section-2", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-3", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-4", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-5", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-6", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-7", list_item_id=None, is_complete=None
+            ),
+        ],
+    )
+
+    questionnaire_store_updater.update_progress_for_dependent_sections()
+
+    assert len(questionnaire_store_updater.evaluated_dependents) == 6
```

Comment:
We need to assert the number of calls to routing path and they are in the right order.
This would pass if any 6 were evaluated even if it was not in the right order. The order and number of calls to routing_path() if the main aim of the test.

We only add evaluated dependants to `evaluated_dependents` if it is not already in `evaluated_dependents` but that doesn't mean it wasn't already evaluated but not added, hence this test wouldn't catch it.

Reply:
I have tried to update this but not confident it is being ordered correctly in the test output based on my mocking at the moment

Reply:
Ya, it wasn't quite right, see: https://github.com/ONSdigital/eq-questionnaire-runner/pull/1044/files#r1196823402 for the fix.